File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPBedSpawnLocation.java
package org.noxet.noxetserver.playerstate.properties;

import org.bukkit.Location;
import org.bukkit.entity.Player;
import org.noxet.noxetserver.playerstate.PlayerStateProperty;

public class PSPBedSpawnLocation implements PlayerStateProperty<Location> {
    @Override
    public String getConfigName() {
        return "bed_spawn_location";
    }

    @Override
    public Location getDefaultSerializedProperty() {
        return null;
    }

    @Override
    public Location getSerializedPropertyFromPlayer(Player player) {
        return player.getBedSpawnLocation();
    }

    @Override
    public void restoreProperty(Player player, Location spawnLocation) {
        player.setBedSpawnLocation(spawnLocation);
    }

    @Override
    public Class<Location> getTypeClass() {
        return Location.class;
    }
}

File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPBedSpawnLocation.scala
package org.noxet.noxetserver.playerstate.properties

import org.bukkit.Location
import org.bukkit.entity.Player
import org.noxet.noxetserver.playerstate.PlayerStateProperty

object PSPBedSpawnLocation extends PlayerStateProperty[Location]:
  override def getConfigName: String = "bed_spawn_location"

  override def getDefaultSerializedProperty: Location = null

  override def getSerializedPropertyFromPlayer(player: Player): Location =
    player.getBedSpawnLocation

  override def restoreProperty(player: Player, spawnLocation: Location): Unit =
    player.setBedSpawnLocation(spawnLocation)